script.js

```diff
@@ -0,0 +1,356 @@
+    this._fireEvent('focus', {
+      itemElem: itemElem,
+      nodeElem: nodeElem,
+      index: Array.prototype.indexOf.call(this.children, nodeElem)
+    });
+
+    // Since we may have many elements in one node, node-changed event happens
+    // when active node is changed.
+    if (this.currentNode && this.currentNode !== nodeElem) {
+      this.currentNode.classList.remove('active');
+      this._fireEvent('node-changed', {
+        scrollbox: this,
+        oldNodeElem: this.currentNode,
+        newNodeElem: nodeElem
+      });
+    }
+    this.currentNode = nodeElem;
+    this.currentNode.classList.add('active');
+    itemElem.focus();
+  };
+
+  /**
+   * Handle unfocus from spacial navigator
+   */
+  proto._handleUnfocus = function(itemElem) {
+    var nodeElem = this._getNodeFromItem(itemElem);
+    this._fireEvent('unfocus', {
+      itemElem: itemElem,
+      nodeElem: nodeElem,
+      index: Array.prototype.indexOf.call(this.children, nodeElem)
+    });
+    itemElem.blur();
+  };
+
+  /**
+   * Add/Remove elements to/from spacial navigator when a child is
+   * added/removed.
+   */
+  proto._handleMutation = function(mutations) {
+    mutations.forEach(function(mutation) {
+      var prevElem = mutation.previousSibling;
+      var nextElem = mutation.nextSibling;
+      if (prevElem) {
+        prevElem = (prevElem.nodeType === Node.ELEMENT_NODE)?
+                                    prevElem : prevElem.previousElementSibling;
+      }
+      if (nextElem) {
+        nextElem = (nextElem.nodeType === Node.ELEMENT_NODE)?
+                                    nextElem : nextElem.nextElementSibling;
+      }
+      var navigableElems;
+      var nodes;
+      if (mutation.type === 'childList') {
+        nodes = Array.prototype.slice.call(mutation.addedNodes);
+        // add every element to spacial navigator
+        nodes.forEach(function(node) {
+          navigableElems = Array.prototype.slice.call(
+                                    node.getElementsByClassName('navigable'));
+          if (node.classList.contains('navigable')) {
+            navigableElems.push(node);
+          }
+          this._spatialNavigator.multiAdd(navigableElems);
+          this._fireEvent('node-added', {
+            node: node
+          });
+        }, this);
+
+        nodes = Array.prototype.slice.call(mutation.removedNodes);
+        // remove every element from spacial navigator
+        nodes.forEach(function(node) {
+          navigableElems = Array.prototype.slice.call(
+                                    node.getElementsByClassName('navigable'));
+          if (node.classList.contains('navigable')) {
+            navigableElems.push(node);
+          }
+          this._spatialNavigator.multiRemove(navigableElems);
+          // If the removed node is active, then we have transfer the active
+          // node to the nearest node, otherwise, adjust the scroll offset
+          if (node.classList.contains('active')) {
```

Comment:
May we use `return this._spatialNavigator.getFocusedElement()` to find the active one instead of checking class list?